src/expr/src/scalar/func.rs

```diff
@@ -2521,6 +2534,9 @@ impl BinaryFunc {
             BinaryFunc::DateTruncTimestamp => {
                 eager!(|a, b: Datum| date_trunc(a, b.unwrap_timestamp()))
             }
+            BinaryFunc::DateTruncInterval => {
+                eager!(|a, b: Datum| date_trunc_interval(a, b.unwrap_interval()))
```

Comment:
I would expect this to simply look like `eager!(date_trunc_interval)`; the surrounding functions have the implementations they do because they're generic, but this is not.

Reply:
Makes sense. I blindly copied the invocation above, without realizing its generic.